client/classic/upe/payment-processing.js

```diff
@@ -424,20 +424,34 @@ export const confirmWalletPayment = async ( api, jQueryForm ) => {
 	}
 
 	const paymentMethodType = partials[ 2 ];
+	const returnURL = decodeURIComponent( partials[ 4 ] );
 
 	try {
 		// Confirm the payment to tell Stripe to display the modal to the customer.
 		let confirmPayment;
-		if ( paymentMethodType === 'wechat_pay' ) {
-			confirmPayment = await api
-				.getStripe()
-				.confirmWechatPayPayment( clientSecret, {
-					payment_method_options: {
-						wechat_pay: {
-							client: 'web',
+		switch ( paymentMethodType ) {
+			case 'wechat_pay':
+				confirmPayment = await api
+					.getStripe()
+					.confirmWechatPayPayment( clientSecret, {
+						payment_method_options: {
+							wechat_pay: {
+								client: 'web',
+							},
 						},
-					},
-				} );
+					} );
+				break;
+			case 'cashapp':
+				confirmPayment = await api
+					.getStripe()
+					.confirmCashappPayment( clientSecret, {
+						return_url: returnURL,
+					} );
+				break;
+			default:
+				// eslint-disable-next-line no-console
+				console.error( 'Invalid wallet type:', paymentMethodType );
+				throw new Error( getStripeServerData()?.invalid_wallet_type );
```

Comment:
I think my preference of this approach was because there are a bunch of localised messages that are listed in 1 place and filtered via the `WC_Stripe_Helper::get_localized_messages()` function so I added this new one there too. 

I don't know if that's something we generally want to move away from but figured keeping it consistent was the way to go for now.